### DISCUSSION TITLE
how to get stream of response by chunks without waiting when response done?

### DISCUSSION THREAD

**Discussion by AndrewBPC** [tectalichq/public-openai-client-php#23] [Q&A] [answered]
Like in https://chat.openai.com/
Or this package don't support steam?

**Comment by the-csaba** [comment, marked as answer]
Hi @AndrewBPC 

You can set the `stream` parameter on the `completions` endpoint.

https://platform.openai.com/docs/api-reference/completions/create

When you specify `stream=true` in your request, you would then need to retrieve the response using the `getResponse()` method rather than `toModel()` or `toArray()`.

You can then access the stream interface and loop over it, outputting it in chunks.

Unfortunately, we don't have any detailed documentation on that, but hopefully, this points you in the right direction.
